modules/exploits/multi/browser/chrome_simplifiedlowering_overflow.rb

```diff
@@ -0,0 +1,172 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ManualRanking
+
+  include Msf::Post::File
+  include Msf::Exploit::Remote::HttpServer
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Google Chrome versions before 87.0.4280.88 integer overflow during SimplfiedLowering phase',
+        'Description' => %q{
+          This module exploits an issue in Google Chrome versions before 87.0.4280.88 (64 bit).
+          The exploit makes use of a integer overflow in the SimplifiedLowering phase in turbofan.
+          It is used along with a typer hardening bypass using ArrayPrototypeShift to create a JSArray with a length of -1.
+          This is abused to gain arbitrary read/write into the isolate region.
+          Then an ArrayBuffer can be used to achieve absolute arbitrary read/write.
+          The exploit then uses WebAssembly in order to allocate a region of RWX memory, which is then replaced with the payload shellcode.
+          The payload is executed within the sandboxed renderer process, the browser must be run with the --no-sandbox option for the payload to work correctly.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Rajvardhan Agarwal (r4j)', # exploit
+        ],
+        'References' => [
+          ['CVE', '2020-16040'],
+          ['URL', 'https://chromium-review.googlesource.com/c/v8/v8/+/2557498'],
+          ['URL', 'https://github.com/r4j0x00/exploits/tree/master/CVE-2020-16040'],
```

Comment:
You might want to add https://bugs.chromium.org/p/chromium/issues/detail?id=1150649 and https://faraz.faith/2021-01-07-cve-2020-16040-analysis/ also